### PR TITLE
Allow derived access chain without uses in access chain conversion

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

**Comment by antiagainst** [comment]
Would appreciate @greg-lunarg 's review too

File: source/opt/local_access_chain_convert_pass.cpp [hunk 1]
```diff
@@ -139,7 +139,17 @@ bool LocalAccessChainConvertPass::IsConstantIndexAccessChain(
...
+  if (!uses) {
+    // This is a variable (or access chain to a variable) that has no uses.
+    // We won't encounter loads or stores for this <result-id> per se, but
+    // this <result-id> may be derived from some other variable (or access
+    // chain). Return true here can unblock the access chain conversion of
+    // the root variable. This particular <result-id> won't be touched and
+    // can be handled in dead code elimination.
+    return true;
+  }
+
```
Comment (dnovillo) [inline source/opt/local_access_chain_convert_pass.cpp]: This should not be needed after @alan-baker PR #987 is in.  @alan-baker could you confirm?

**Comment by antiagainst** [comment]
Oh, forgot that this function is duplicated at two other places... Will add them too.

File: source/opt/local_access_chain_convert_pass.cpp (hunk 1 above)
Comment (dneto0) [reply source/opt/local_access_chain_convert_pass.cpp]: Agree.  #987 changes the API to provide a ForEachUse or ForEachUser and you can't get nullptr at all.

Comment (antiagainst) [reply source/opt/local_access_chain_convert_pass.cpp]: Oh, great! That would be better than this ad-hoc fix.

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as b02c9a58

**Comment by antiagainst** [comment]
Thanks, @dneto0! :)

**Pull request closed without merging**